### ISSUE TITLE
Flaky image loading when using css content

### ISSUE THREAD

**Issue by StephenLarkin** [simonhaenisch/md-to-pdf#27]
I have a series of documents that we're using for our markdown documentation. Each of them has a title page with a header and footer, which I'm injecting in using the css content property to avoid duplication.

The header contains an image of our company logo, but I've been finding that when I compile the file, sometimes it doesn't load in, leaving just a blank space at the top.

```css
&::before {
  content: url("img/logo.svg");
  display: block;
  height: 80px;
}
```

I can only assume that, as it works perfectly fine around half of the time, that there is some sort of race condition between loading the page in chromium headless and printing out to the pdf.

**Comment by simonhaenisch** [comment]
Thanks for the info. I implemented some code that waits till there are no more requests using the `request` and `requestfinished`/`requestfailed` events, but maybe that request doesn't trigger a `request` event on the page because it comes from a stylesheet... need to investigate that first.

Actually just looking into `write-pdf.js`, I think there might be a better way to instantiate the page (there is a lot of async stuff that can be run in parallel which is currently executed sequentially), which might make the `waitUntil: 'networkidle0' option for puppeteer's `goto` function work properly, and thus makes my custom request waiting code obsolete. Will try now and publish a patch if it works :nerd_face:

**Comment by simonhaenisch** [comment]
@StephenLarkin can you please try with [v2.6.3](https://github.com/simonhaenisch/md-to-pdf/releases/tag/v2.6.3) and let me know if you still get that issue?

**Comment by StephenLarkin** [comment]
@simonhaenisch Thanks for moving quickly on this, but it seems like something in 2.6.3 has broken how the CSS is loaded for one of the dependencies. I can no longer build with this version.

md-to-pdf ThemeGuide.md ThemeGuide.pdf --config-file config.json

  × generating PDF from ThemeGuide.md
    → Loading style from D:\Repos\Folder\docs-md\node_modules\highlight.js\styles\github.css failed
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! docs@1 gen:themes: `md-to-pdf ThemeGuide.md ThemeGuide.pdf --config-file config.json`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the docs@1 gen:themes script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by simonhaenisch** [comment]
@StephenLarkin thanks for the report and sorry about that... there’s another (unrelated) thing that I was working on that seems to not work as expected. I’ll need to figure out how to run puppeteer in the CI so I can have some better testing :nerd_face:

**Comment by simonhaenisch** [comment]
This should be fixed in [v2.6.4](https://github.com/simonhaenisch/md-to-pdf/releases/tag/v2.6.4)?

**Comment by StephenLarkin** [comment]
@simonhaenisch Yep. That seems to have fixed that issue with the flaky loading. It now generates correctly each time.

Thanks again for getting to this so quickly.

I've noticed another issue with it though, but I think we can close out this bug and I'll raise it as another.
